lib/catalog/owid/catalog/utils.py

```diff
@@ -200,3 +202,28 @@ def dynamic_yaml_to_dict(yd: Any) -> dict:
     """Convert dynamic yaml to dict. Using dynamic yaml can cause problems when you
     try to run e.g. Origin(**yd). It's safer to run Origin(**dynamic_yaml_to_dict(yd)) instead."""
     return yaml.safe_load(dynamic_yaml.dump(yd))
+
+
+def hash_any(x: Any) -> int:
+    """Return unique deterministic hash for an arbitrary object. This is useful if you can't make your dataclasses
+    frozen but still want to use operations such as `set` or `unique`.
+
+    The problem with using `hash` function directly is that Python uses random hash seed (PYTHONHASHSEED) for
+    security reasons. So the hash of the same object can be different in different runs for strings. This is
+    why we use md5 hash for strings.
+    """
+    if is_dataclass(x):
```

Comment:
Wow, this function is quite obscure, it would be nice to explain a bit more what's going on.

Reply:
Added some comments.